faq/mpi-legacy.inc

```diff
@@ -0,0 +1,593 @@
+     "MPI_TYPE_CREATE_HINDEXED", "MPI_Type_create_hindexed") .
+mpi1("MPI_TYPE_HVECTOR", "mpi-1-mpi-type-hvector",
+     "MPI_TYPE_CREATE_HVECTOR", "MPI_Type_create_hvector") .
+mpi1("MPI_TYPE_LB", "mpi-1-mpi-type-lb",
+     "MPI_TYPE_GET_EXTENT", "MPI_Type_get_extent") .
+mpi1("MPI_TYPE_STRUCT", "mpi-1-mpi-type-struct",
+     "MPI_TYPE_CREATE_STRUCT", "MPI_Type_create_struct") .
+mpi1("MPI_TYPE_UB", "mpi-1-mpi-type-ub",
+     "MPI_TYPE_GET_EXTENT", "MPI_Type_get_extent") .
+mpi1("MPI_LB", "mpi-1-mpi-lb-ub",
+     "MPI_TYPE_CREATE_RESIZED", "MPI_Type_create_resized") .
+mpi1("MPI_UB", "mpi-1-mpi-lb-ub",
+     "MPI_TYPE_CREATE_RESIZED", "MPI_Type_create_resized") .
+mpi1("MPI_COMBINER_HINDEXED_INTEGER", "mpi-1-mpi-combiner-fortran-integers",
+     "MPI_COMBINER_HINDEXED", "MPI_Type_get_envelope") .
+mpi1("MPI_COMBINER_HVECTOR_INTEGER", "mpi-1-mpi-combiner-fortran-integers",
+     "MPI_COMBINER_HVECTOR", "MPI_Type_get_envelope") .
+mpi1("MPI_COMBINER_STRUCT_INTEGER", "mpi-1-mpi-combiner-fortran-integers",
+     "MPI_COMBINER_STRUCT", "MPI_Type_get_envelope") .
+mpi1("MPI_HANDLER_FUNCTION", "mpi-1-mpi-handler-function",
+     "MPI_COMM_ERRHANDLER_FUNCTION", "MPI_Comm_create_errhandler") .
+"</table></div>
+
+Although these symbols are no longer prototyped in [mpi.h], _they are
+still present in the MPI library in Open MPI v4.0.x._ This enables
+legacy MPI applications to _link and run_ successfully with Open MPI
+v4.0.x, even though they will fail to _compile_.
+
+*The Open MPI team <font color=\"red\">strongly</font> encourages all
+MPI application developers to stop using these constructs that were
+first deprecated over 20 years ago, and finally removed from the MPI
+specification in MPI-3.0 (in 2012).* The FAQ items in this category
+show how to update your application to stop using these legacy
+symbols.
+
+All that being said, if you are unable to immediately update your
+application to stop using these legacy MPI-1 symbols, you can
+re-enable them in [mpi.h] by configuring Open MPI with the
+[--enable-mpi-compatibility] flag.
+
+*NOTE:* Future releases of Open MPI beyond the v4.0.x series may
+remove these symbols altogether.";
+
+/////////////////////////////////////////////////////////////////////////
+
+$current_year = date("Y");
+$years_ago = $current_year - 1996;
+
+$q[] = "Why on earth are you breaking the compilation of MPI
+applications?";
+
+$anchor[] = "why-breaking-legacy-applications";
+
+$a[] = "The Open MPI developer community decided to take a first step
+of removing the prototypes for these symbols from [mpi.h] starting
+with the Open MPI v4.0.x series for the following reasons:
+
+<ol>
+<li> These symbols have been deprecated since *1996.* That's
+<em>$years_ago years ago!</em>  It's time to start raising awareness
+for developers who are inadvertantly still using these legacy
+symbols.</li>
+<li> The MPI Forum removed these symbols from the MPI-3.0
+specification in 2012.  This is a symbol that the Forum itself
```

Comment:
This is a sign that?

Reply:
Wow.  How on earth did that get typed?  Fixed.